paasta_tools/contrib/update_soa_memcpu.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+import argparse
+import contextlib
+import os
+import subprocess
+import sys
+import tempfile
+
+import ruamel.yaml as yaml
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(
+        description='',
+    )
+    parser.add_argument(
+        '-s', '--service',
+        help="Service to edit. Like 'example_service'.",
+        dest="service",
+        required=True,
+    )
+    parser.add_argument(
+        '-i', '--instance',
+        help="Instance of the service to edit. Like 'main' or 'canary'.",
+        dest="instance",
+        required=True,
+    )
+    parser.add_argument(
+        '-c', '--cluster',
+        help="The PaaSTA cluster that has the service instance to edit. Like 'norcal-prod'.",
+        dest="cluster",
+        required=True,
+    )
+    parser.add_argument(
+        '-m', '--mem',
+        default='',
+        dest="mem",
+        help='New value for mem of service',
+    )
+    parser.add_argument(
+        '-cp', '--cpu',
+        default='',
+        dest="cpu",
+        help='New value for cpus of service',
+    )
+    parser.add_argument(
+        '-t', '--ticket',
+        dest="ticket",
+        help='JIRA ticket tracking changes',
+    )
+    return parser.parse_args(argv)
+
+
+def tempdir():
+    return tempfile.TemporaryDirectory(
+        prefix='repo',
+        dir='/nail/tmp',
+    )
+
+
+@contextlib.contextmanager
+def cwd(path):
+    pwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(pwd)
+
+
+@contextlib.contextmanager
+def in_tempdir():
+    with tempdir() as tmp:
+        with cwd(tmp):
+            yield
+
+
+def clone(branch_name):
+    print('cloning')
+    remote = 'git@sysgit.yelpcorp.com:yelpsoa-configs'
+    subprocess.check_call(('chronic', 'git', 'clone', remote, '.'))
```

Comment:
I would prefer you not use chronic. For this kind of tool I think it is ok to just say what it is doing